core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/AbstractSample.java

```diff
@@ -713,20 +748,29 @@ public Sample build() {
       checkArgument(!LimsUtils.isStringEmptyOrNull(scientificName), "Must provide a scientificName to create a Sample");
 
       if (sampleAdditionalInfo == null) {
-        // Classic MISO Sample.
+        log.debug("Create a classic MISO sample.");
         return new SampleImpl(this);
       } else {
-        checkNotNull(parent, "parent must be provided to create a Sample.");
         checkNotNull(sampleAdditionalInfo, "SampleAdditionalInfo must be provided to create a Sample.");
         checkNotNull(sampleAdditionalInfo.getSampleClass(), "SampleAdditionalInfo.sampleClass must be provided to create a Sample.");
         if (identity != null) {
-          // Or we could mark a SampleClass as the root.
+          log.debug("Create an Identity.");
           checkNotNull(rootSampleClass, "A root SampleClass must be provided to create an Identity Sample.");
           return new SampleIdentityNode(this);
         } else if (sampleAnalyte != null) {
+          log.debug("Create an Analyte.");
+          checkNotNull(parent, "parent must be provided to create a Sample.");
           checkNotNull(sampleAdditionalInfo.getTissueOrigin(), "SampleAdditionalInfo.tissueOrigin must be provided to create a Sample.");
           checkNotNull(sampleAdditionalInfo.getTissueType(), "SampleAdditionalInfo.tissueType must be provided to create a Sample.");
-          return new SampleAnalyteNode(this);
+          return SampleImpl.sampleAnalyte(this);
+        } else if (sampleTissue != null) {
+          checkNotNull(sampleTissue, "sample tissue must be provided to create a SampleTissue");
+          checkNotNull(sampleTissue.getSample(), "sample must be provided to create a SampleTissue");
+          checkNotNull(sampleTissue.getLab(), "Lab must be provided to create a SampleTissue");
+          checkNotNull(sampleTissue.getLab().getAlias(), "Lab alias must be provided to create a SampleTissue");
+          checkNotNull(sampleTissue.getLab().getInstitute(), "Institute must be provided to create a SampleTissue");
+          checkNotNull(sampleTissue.getLab().getInstitute().getAlias(), "Institute alias must be provided to create a SampleTissue");
+          return new SampleTissueNode(this);
```

Comment:
My understanding of Lab and Institute is that Lab alias, Lab Institute, and Institute Alias are all admin-only for create and edit, and the null checks are enforced by the database. If that's indeed the case, then shouldn't the user be providing only a Lab? (This would also mean that the user can't actually fix any errors for null lab alias, lab institute or lab institute alias)